content/doc/book/using/using-agents.adoc

```diff
@@ -0,0 +1,195 @@
+---
+layout: section
+title: Using Jenkins agents
+---
+ifdef::backend-html5[]
+:description:
+:author:
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+:toc:
+:toclevels: 4
+:hide-uri-scheme:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+endif::[]
+
+The Jenkins architecture is designed for distributed build environments.
+It allows us to use different environments for each build project balancing
+the workload among multiple agents running jobs in parallel.
+
+The Jenkins controller is the orignal node in the Jenkins installation.
+The Jenkins controller administers the Jenkins agents and orchestrates their work, including scheduling jobs on agents and monitoring agents.
+Agents may be connected to the Jenkins controller using either local or cloud computers.
+
+The agents require a Java installation and a network connection to the Jenkins controller.
+
+== Configuring agents with Docker
+
+Jenkins agents can be launched in physical machines, virtual machines, Kubernetes clusters, and with Docker images.
+This section connects Docker agents to Jenkins with SSH.
+
+=== Environment
+
+To run this guide you will need a machine with:
+
+* Java installation
+* Jenkins installation
+* Docker installation
+* SSH key pair
+
+[NOTE]
+====
+If you need help to install Java, Jenkins and Docker please visit the section link:/doc/book/installing/docker/[Installing Jenkins.]
+====
+
+=== Generating an SSH key pair
+
+1. In a terminal window run the command: `ssh-keygen -f ~/.ssh/jenkins_agent_key`
+2. Provide a passphrase to use with the key
+3. Confirm the output looks something like this:
++
+[source,bash]
+----
+ubuntu@desktop:~$ ssh-keygen -f ~/.ssh/jenkins_agent_key
+Generating public/private rsa key pair.
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in /home/ubuntu/.ssh/jenkins_agent_key
+Your public key has been saved in /home/ubuntu/.ssh/jenkins_agent_key.pub
+The key fingerprint is:
+SHA256:XqxxjqsLlvDD0ZHm9Y2iR7zC6IbsUlMEHo3ffy8TzGs
+The key's randomart image is:
++---[RSA 3072]----+
+|  o+             |
+| ...o  .         |
+|  .o .+ .        |
+|    o+.+ o o     |
+|  ... o.So* .    |
+|  o+ = +.X=      |
+| o oO + *..+     |
+|. oo.o o .E .    |
+| o... oo.. o     |
++----[SHA256]-----+
+----
+
+==== Create a Jenkins SSH credential
+
+1. Go to your Jenkins dashboard;
+2. Go to `Manage Jenkins` option in main menu and click on credentials button;
++
+image:node/credentials-1.png[credentials menu]
```

Comment:
Limit width of image, allow user to open full size image in another window or tab
```suggestion
image:node/credentials-1.png[credentials menu,700]
```